flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/expressions/comparison.scala

```diff
@@ -74,6 +109,8 @@ case class IsNull(child: Expression) extends UnaryExpression {
   override def toRexNode(implicit relBuilder: RelBuilder): RexNode = {
     relBuilder.isNull(child.toRexNode)
   }
+
+  override def resultType = BOOLEAN_TYPE_INFO
```

Comment:
Oh yes, sure. You're right. I did not expand the code (Github had folded some lines) I thought this belonged to `GreaterThan`.